docs/taskruns.md

```diff
@@ -876,11 +876,41 @@ TaskRuns can be halted on failure for troubleshooting by providing the following
 ```yaml
 spec:
   debug:
-    breakpoint: ["onFailure"]
+    breakpoint: 
+      onFailure: true
 ```
 
-Upon failure of a step, the TaskRun Pod execution is halted. If this TaskRun Pod continues to run without any lifecycle
-change done by the user (running the debug-continue or debug-fail-continue script) the TaskRun would be subject to
+### Breakpoint before step
+
+If you want to set a breakpoint before the step is executed, you can add the step name to the `beforeSteps` field in the following way:
+
+```yaml
+spec:
+  debug:
+    breakpoints:
+      beforeSteps: 
+        - {{ stepName }}
+```
+
+### Breakpoint after step
+
+`after step` will take effect when the commands are executed without error, 
+if you want to set a breakpoint after the step is executed, you can add the step name to the `beforeSteps` field in the following way:
+
+```yaml
+spec:
+  debug:
+    breakpoints:
+      afterSteps: 
+        - {{ stepName }}
+```
+
+### How to enter container debugging
+
+Upon any breakpoints of a step, the TaskRun Pod execution is halted. If this TaskRun Pod continues to run without any lifecycle
+change done by the user (running the debug-continue or debug-fail-continue script or debug-beforestep-continue
+or debug-beforestep-fail-continue or debug-afterstep-continue or debug-afterstep-fail-continue)
+the TaskRun would be subject to
```

Comment:
The command below is missing the container command, could you fix that?

```bash
kubectl exec -it print-date-d7tj5-pod -c step-print-date-human-readable -- bash
```

Reply:
I have fixed the missing `-- bash` command